addons/website_slides/controllers/main.py

```diff
@@ -624,20 +624,24 @@ def slide_quiz_get(self, slide_id):
             return fetch_res
         slide = fetch_res['slide']
         quiz_info = self._get_slide_quiz_info(slide)
-        return {
+        quiz_data = {
             'questions': [{
                 'id': question.id,
                 'question': question.question,
                 'answers': [{
                     'id': answer.id,
                     'text_value': answer.text_value,
-                    'is_correct': answer.is_correct,
                 } for answer in question.answer_ids],
             } for question in slide.question_ids],
             'quizAttemptsCount': quiz_info['quiz_attempts_count'],
             'quizKarmaGain': quiz_info['quiz_karma_gain'],
             'quizKarmaWon': quiz_info['quiz_karma_won'],
         }
+        if slide.user_membership_id.sudo().completed:
+            for index_question, question in enumerate(slide.question_ids):
+                for index_answer, answer in enumerate(question.answer_ids):
+                    quiz_data['questions'][index_question]['answers'][index_answer]['is_correct'] = answer.is_correct
+        return quiz_data
```

Comment:
I would rather add a variable to get once is slide is completed and use it directly in the returned dict: 
```
slide_is_completed = slide.user_membership_id.sudo().completed
quiz_data = {
            'questions': [{
                'id': question.id,
                'question': question.question,
                'answers': [{
                    'id': answer.id,
                    'text_value': answer.text_value,
                    'is_correct': answer.is_correct and slide_is_completed,
                } for answer in question.answer_ids],
            } for question in slide.question_ids],
            'quizAttemptsCount': quiz_info['quiz_attempts_count'],
            'quizKarmaGain': quiz_info['quiz_karma_gain'],
            'quizKarmaWon': quiz_info['quiz_karma_won'],
        }
```
But.. I don't really understand why this change ? what's the impact if we don't change this at all ?
Is this for rpc security information leak reasons ? -> If yes, write it in the code. Aaaaand, using 'my' method will return false for every answer. So that the hacker is screwed anyway :D

Reply:
JEM thought it was better to not have the key in the dict if the slide wasn't completed by the user. And yes, the change was made to prevent users to cheat the quiz.

Reply:
try `None if not completed else answer.is_correct`
if None is transformed into undefined, then okay to do that.

Reply:
Soooo, as discussed in a wonderful team brainstorming, we will go for 
`'is_correct': answer.is_correct if slide_is_completed else None.`
So that the variable is not returning any real value that could be misused by confusion